Mathlib/CategoryTheory/Bicategory/Opposites.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2025 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.Bicategory.Basic
+
+/-!
+# Opposite bicategories
+
+We construct the 1-cell opposite of a bicategory `B`, called `Bᴮᵒᵖ`. It is defined as follows
+* The objects of `Bᴮᵒᵖ` correspond to objects of `B`.
+* The morphisms `X ⟶ Y` in `Bᴮᵒᵖ` are the morphisms `Y ⟶ X` in `B`.
+* The 2-morphisms `f ⟶ g` in `Bᴮᵒᵖ` are the 2-morphisms `f ⟶ g` in `B`. In other words, the
+  directions of the 2-morphisms are preserved.
+
+Note that the standard notation for the opposite of a bicategory is `Bᵒᵖ`, however this clashes
+with the notation for the opposite of a 1-category, so we use `Bᴮᵒᵖ` instead.
+
+# Remarks
+There are multiple notions of opposite categories for bicategories.
+- There is 1-cell dual `Bᴮᵒᵖ` as defined above.
+- There is the 2-cell dual, `Cᶜᵒ` where only the natural transformations are reversed
+- There is the bi-dual `Cᶜᵒᵒᵖ` where the directions of both the morphisms and the natural
+  transformations are reversed.
+
+## TODO
+
+* Define the 2-cell dual `Cᶜᵒ`.
+* Provide various lemmas for going between `LocallyDiscrete Cᵒᵖ` and `(LocallyDiscrete C)ᵒᵖ`.
+
+Note: `Cᶜᵒᵒᵖ` is WIP by Joël Riou and Christian Merten.
+
+-/
+
+universe w v u
+
+open CategoryTheory Bicategory Opposite
+
+
+/-- The type of objects of the 1-cell opposite of a bicategory `B` -/
+structure Bicategory.Opposite (B : Type u) where
+  /-- The object of `Bᴮᵒᵖ` that represents `b : B` -/  bop ::
+  /-- The object of `B` that represents `b : Bᴮᵒᵖ` -/ unbop : B
+
+namespace Bicategory.Opposite
+
+variable {B : Type u}
+
+@[inherit_doc]
+notation:max B "ᴮᵒᵖ" => Bicategory.Opposite B
```

Comment:
It seems to me we could use `ᵒᵖ` for the `1`-cell opposite. In order to do that, we should probably add a `CategoryStruct` instance on `Cᵒᵖ` in `CategoryTheory.Opposites` before defining the opposite category.